cheroot/workers/monitor.py

```diff
@@ -0,0 +1,154 @@
+"""Monitor for threadpool."""
+__metaclass__ = type
+
+import threading
+import time
+
+from .dynpool import DynamicPoolResizer
+
+
+class BackgroundTask(threading.Thread):
+    """A subclass of threading.Thread whose run() method repeats.
+
+    Use this class for most repeating tasks. It uses time.sleep() to wait
+    for each interval, which isn't very responsive; that is, even if you call
+    self.cancel(), you'll have to wait until the sleep() call finishes before
+    the thread stops. To compensate, it defaults to being daemonic, which means
+    it won't delay stopping the whole process.
+    """
+
+    def __init__(
+        self,
+        interval,
+        function,
+        args=None,
+        kwargs=None,
+        logger=None,
+    ):
+        """Initialize a BackgroundTask."""
+        super(BackgroundTask, self).__init__()
+        self.interval = interval
+        self.function = function
+        self.args = args if args is not None else []
+        self.kwargs = kwargs if kwargs is not None else {}
+        self.running = False
+        self.log = logger or (lambda msg: None)
+        # default to daemonic
```

Comment:
This is too obvious to make it to a comment. Comments must explain why and not rephrase the code exactly as it reads.

Reply:
removed